src/components/Page.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent, HTMLAttributes, useEffect, useRef } from 'react';
+
+import viewManager from './viewManager/viewManager';
+
+type PageProps = {
+    title?: string,
+    isBackButtonEnabled?: boolean
+};
+
+/**
+ * Page component that handles hiding active non-react views, triggering the required events for
+ * navigation and appRouter state updates, and setting the correct classes and data attributes.
+ */
+const Page: FunctionComponent<PageProps & HTMLAttributes<HTMLDivElement>> = ({
+    children,
+    className = '',
+    title,
+    isBackButtonEnabled = true
+}) => {
+    const element = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        // hide active non-react views
+        viewManager.hideView();
+    }, []);
+
+    useEffect(() => {
+        const event = {
+            bubbles: true,
+            cancelable: false,
+            detail: {
+                isRestored: false
+            }
+        };
+        // pagebeforeshow - hides tabs on tabless pages in libraryMenu
+        element.current?.dispatchEvent(new CustomEvent('pagebeforeshow', event));
+        // viewshow - updates state of appRouter
```

Comment:
`viewbeforeshow` need here for theme

Reply:
Good catch! I added the event as well as some additional values in the event details that were required for theme songs/videos and the now playing bar to work correctly.